test/async/retry_test.dart

```diff
@@ -46,5 +48,12 @@ void main() {
               timeout: const Duration(milliseconds: 1)),
           throwsA(equals('error')));
     });
+
+    test('should timeout long tasks', () async {
+      var task = () => new Completer().future;
+
+      expect(retry(task, timeout: const Duration(milliseconds: 100)),
```

Comment:
You should also look at using FakeAsync when writing tests that depend on time, granted your test here is perfectly OK because the "task" is always going to wait forever.